### DISCUSSION TITLE
whisperx implementation

### DISCUSSION THREAD

**Discussion by farzadnk** [absadiki/subsai#37] [Ideas]
Hi, you should think about whisperx implementation. is have the best timestamps right now.
https://github.com/m-bain/whisperX

**Comment by absadiki** [comment]
Hi @farzadnk,

Is `whisperX` better now than `whisper_timestamps` ? Have you tried it ?

**Reply by farzadnk** [reply]
hi @abdeladim-s 
I tried whisper_timestamps but it's not as accurate as WhisperX, sometimes the words are not in sync and overall WhisperX has better capabilities.


You can see my result here
whisper_timestamps :
https://github.com/abdeladim-s/subsai/assets/132581088/8905f605-8b0c-4ae5-8188-739f2e561574

whisperX:
https://github.com/abdeladim-s/subsai/assets/132581088/ae6e584a-708c-4260-bd93-cdd69c788535

**Reply by absadiki** [reply]
Yeah you are right, `whisperX` seems really accurate. I think I will add it then. 
I will let you know once it is finished to give it a try. 
Thanks @farzadnk.

**Reply by farzadnk** [reply]
Thanks  @abdeladim-s  I really appreciate it

**Comment by absadiki** [comment]
Hi @farzadnk, whisperX support has been added as requested. 
Hope it will be useful :)

**Reply by farzadnk** [reply]
Hi @abdeladim-s 

thanks a lot man, how to get **word.srt** (**word-timestamps**)output?

I use this method, but there seems to be no other options

```
subs = pysubs2.load_from_whisper(result)
subs.save('name.srt')

```

**Reply by absadiki** [reply]
Yeah sorry @farzadnk, I forget to add the `segment_type` option to get word level timestamps. 
I have added it now to the [WhisperX configs](https://abdeladim-s.github.io/subsai/#subsai.models.whisperX_model.WhisperXModel.config_schema). 
Please give it a try now.
